examples/benchmark/src/query.rs

```diff
@@ -0,0 +1,149 @@
+                    Err(e) => {
+                        _ = query.reply_tx.send_error(&e.to_string());
+                    }
+                }
+            }
+            score = score_rx.recv() => {
+                let Some(score) = score else {
+                    info!("score channel closed");
+                    break;
+                };
+                state.score = score;
+            }
+        }
+    }
+}
+
+async fn score_update(tx: Sender<BenchScore>) {
+    debug!("score update task started");
+    loop {
+        let net_start_time = net_now().await;
+        let local_start_time = Instant::now();
+        let (gas_at_start, _) =
+            wapo::ocall::app_gas_consumed().expect("failed to get gas consumed");
+        debug!("net_start_time: {:?}", net_start_time);
+        debug!("gas_at_start: {:?}", gas_at_start);
+        wapo::time::sleep(Duration::from_secs(30)).await;
+        let (gas_at_end, token) =
+            wapo::ocall::app_gas_consumed().expect("failed to get gas consumed");
+        let local_elapsed = local_start_time.elapsed();
+        let net_end_time = net_now().await;
+        debug!("net_end_time: {:?}", net_end_time);
+        debug!("gas_at_end: {:?}", gas_at_end);
+        match (&net_start_time, &net_end_time) {
+            (Ok(start_time), Ok(end_time)) => {
+                let Ok(net_elapsed) = end_time.duration_since(*start_time) else {
+                    info!("invalid net time, skipping score update");
+                    continue;
+                };
+                let diff = local_elapsed.as_secs_f64() - net_elapsed.as_secs_f64();
+                if diff.abs() > 10_f64 {
+                    info!("time diff between local and net is too large: {diff}");
+                    continue;
+                }
+                let gas_diff = gas_at_end.saturating_sub(gas_at_start);
+                let score = gas_diff.saturating_div(local_elapsed.as_secs());
+                debug!("score: {:?}", score);
+                let score = BenchScore {
+                    gas_per_second: score,
+                    gas_consumed: gas_at_end,
+                    timestamp_secs: end_time
+                        .duration_since(SystemTime::UNIX_EPOCH)
+                        .expect("failed to get timestamp")
+                        .as_secs(),
+                    metrics_token: token,
+                };
+                tx.send(score).await.expect("failed to send score");
+            }
+            _ => {
+                info!("failed to get net time, skipping score update");
+            }
+        }
+    }
+}
+
+async fn net_now() -> Result<SystemTime> {
+    let servers = [
+        "https://www.cloudflare.com",
+        "https://www.apple.com",
+        "https://www.baidu.com",
+        "https://kernel.org/",
+    ];
+
+    let mut futures = FuturesUnordered::new();
+    for server in servers {
+        futures.push(httptime::get_time(server, Duration::from_secs(2)));
```

Comment:
This fetch timestamp from the internet via the HTTP response header `Date: xxxx-xx-xx...`.